### PR TITLE
add WeightOnlyLinear for low memory inference.

### PR DESCRIPTION
**Pull request by xin3he**
## Type of Change

feature

## Description

add WeightOnlyLinear for weight-only approach to allow low memory inference.

## Expected Behavior & Potential Risk

UT pass

## How has this PR been tested?

local test

## Dependency Change?

N/A

### CODE REVIEW

**Comment by xin3he** [comment]
Pre-ci failed due to random issue, reviewer can skip it.

File: neural_compressor/adaptor/torch_utils/model_wrapper.py [hunk 1]
```diff
@@ -146,3 +148,146 @@ def _wrapper_qdq_linear(tmp_model, module_name_list=[]):
...
+            'scale', 
+            torch.zeros(
+                (out_features, math.ceil(in_features / self.groupsize)), 
+                dtype=torch.float,
+            )
+        )
+
+    def pack(self, int_weight, scale, zp, bias):
+        if bias is not None:
+            self.register_buffer('bias', torch.zeros(self.out_features, dtype=torch.float))
```
Comment (wenhuach21) [inline neural_compressor/adaptor/torch_utils/model_wrapper.py:179]: have you copied the bias?

File: test/quantization/test_weight_only_quantization.py [hunk 2]
```diff
@@ -52,6 +53,7 @@ def test_trace(self):
     def test_rtn(self):
         fp32_model = copy.deepcopy(self.model)
```
Comment (wenhuach21) [inline test/quantization/test_weight_only_quantization.py:54]: the ut should cover linear layer with non-zero bias

File: neural_compressor/adaptor/torch_utils/weight_only.py [hunk 3]
```diff
@@ -323,6 +372,8 @@ def awq_quantize(model, weight_config={}, absorb_dict={}, dataloader=None, n_sam
         mse_range (bool, optional):  whether enable clip for weight by checking mse. Defaults to True.
         calib_func: a custom inference function to replace dataloader and iters.
         n_blocks: split model into block number to avoid OOM.
+        return_int (bool, optional): Choose return fp32 or int32 model.
```
Comment (hshen14) [inline neural_compressor/adaptor/torch_utils/weight_only.py:375]: People may not understand why INT32 model. How about reusing export function to save INT4 model (avoid many functionalities inside **quantize**)?

**Review by wenhuach21 — APPROVED** [review]
need to figure out how to fix double quantization issue of sym schema later

**Pull request merged**